signalr-client-sdk/src/main/java/microsoft/aspnet/signalr/client/hubs/HubConnection.java

```diff
@@ -74,7 +75,8 @@ public void onReceived(JsonElement message) {
         super.onReceived(message);
 
         log("Processing message", LogLevel.Information);
-        if (getState() == ConnectionState.Connected) {
+        ConnectionState state = getState();
+        if (state == ConnectionState.Connected || (getTransportClass().equals(LongPollingTransport.class) && (state == ConnectionState.Connected || state == ConnectionState.Reconnecting))) {
```

Comment:
You should call `equals` on `LongPollingTransport.class` at first to avoid NPE
```java
final ConnectionState state = getState();
if (state == ConnectionState.Connected || (LongPollingTransport.class.equals(getTransportClass()) && state == ConnectionState.Reconnecting)) {
```